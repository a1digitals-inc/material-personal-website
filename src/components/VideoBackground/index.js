import React, { useState, useLayoutEffect } from "react";
import PropTypes from "prop-types";

import useStyles from "./styles";
import videPoster240 from "../../assets/images/code240.jpg";
import videPoster320 from "../../assets/images/code320.jpg";
import videPoster480 from "../../assets/images/code480.jpg";
import videPoster720 from "../../assets/images/code720.jpg";
import videoSource240 from "../../assets/videos/code240.mp4";
import videoSource320 from "../../assets/videos/code320.mp4";
import videoSource480 from "../../assets/videos/code480.mp4";
import videoSource720 from "../../assets/videos/code720.mp4";

const VideoBackground = ({ children }) => {
  const classes = useStyles();

  const [videPoster, setVidePoster] = useState(videPoster240);
  const [videoSource, setVideoSource] = useState(videoSource240);

  useLayoutEffect(() => {
    const { innerWidth: width } = window;

    if (width > 1280) {
      setVidePoster(videPoster720);
      setVideoSource(videoSource720);
    } else if (width > 720) {
      setVidePoster(videPoster480);
      setVideoSource(videoSource480);
    } else if (width > 480) {
      setVidePoster(videPoster320);
      setVideoSource(videoSource320);
    } else {
      setVidePoster(videPoster240);
      setVideoSource(videoSource240);
    }
  }, []);

  return (
    <>
      <video
        playsInline
        autoPlay
        loop
        poster={videPoster}
        className={classes.video}
      >
        <source src={videoSource} type="video/mp4" />
      </video>
      <div className={classes.overlay} />
      <div className={classes.content}>{children}</div>
    </>
  );
};

VideoBackground.propTypes = {
  children: PropTypes.node.isRequired,
};

export default VideoBackground;
