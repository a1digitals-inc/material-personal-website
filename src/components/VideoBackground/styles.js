import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles({
  video: {
    position: "fixed",
    right: "0",
    bottom: "0",
    minWidth: "100%",
    minHeight: "100%",
    margin: "0",
    padding: "0",
    overflow: "hidden",
  },
  overlay: {
    position: "fixed",
    right: "0",
    bottom: "0",
    minWidth: "100%",
    minHeight: "100%",
    margin: "0",
    padding: "0",
    overflow: "hidden",
    zIndex: "1",
    backgroundImage: "linear-gradient(to right, #cdcdcd, #f5f5f5)",
    opacity: "0.75",
  },
  content: {
    zIndex: "2",
    position: "relative",
  },
});

export default useStyles;
