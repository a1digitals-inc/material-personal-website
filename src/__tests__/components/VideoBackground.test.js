import React from "react";
import { shallow } from "enzyme";

import VideoBackground from "../../components/VideoBackground";
import useBackgroundVideo from "../../hooks/useBackgroundVideo";
import videPoster240 from "../../assets/images/code240.jpg";
import videoSource240 from "../../assets/videos/code240.mp4";

jest.mock("../../hooks/useBackgroundVideo");

it("renders without crashing", () => {
  useBackgroundVideo.mockImplementation(() => {
    return {
      videPoster: videPoster240,
      videoSource: videoSource240,
    };
  });

  const wrapper = shallow(
    <VideoBackground>
      <div className="unique" />
    </VideoBackground>
  );

  expect(wrapper).toMatchSnapshot();
  expect(wrapper.find("video").prop("poster")).toEqual(videPoster240);
  expect(wrapper.find("source").prop("src")).toEqual(videoSource240);
});
