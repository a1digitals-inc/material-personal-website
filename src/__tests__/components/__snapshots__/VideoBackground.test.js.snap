// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<Fragment>
  <video
    autoPlay={true}
    className="makeStyles-video-1"
    loop={true}
    playsInline={true}
    poster="videopostertest.jpg"
  >
    <source
      src="videosourcetest.mp4"
      type="video/mp4"
    />
  </video>
  <div
    className="makeStyles-overlay-2"
  />
  <div
    className="makeStyles-content-3"
  >
    <div
      className="unique"
    />
  </div>
</Fragment>
`;
